commit 4e57ef78e14aca718e0239e202280c0937f187ff
Author: Matthew Glazar <strager.nds@gmail.com>
Date:   Tue Nov 19 22:05:46 2019 -0800

    open Poly

diff --git a/lib/configuration/command_configuration.ml b/lib/configuration/command_configuration.ml
index ecd2c92..0ddf794 100644
--- a/lib/configuration/command_configuration.ml
+++ b/lib/configuration/command_configuration.ml
@@ -1,6 +1,7 @@
 open Core
 
 open Language
+open Poly
 
 let debug =
   Sys.getenv "DEBUG_COMBY"
diff --git a/lib/match/match_context.ml b/lib/match/match_context.ml
index a2f9cc8..bb57d38 100644
--- a/lib/match/match_context.ml
+++ b/lib/match/match_context.ml
@@ -1,4 +1,5 @@
 open Core
+open Poly
 
 type t =
   { range : Range.t
diff --git a/lib/matchers/alpha/matcher.ml b/lib/matchers/alpha/matcher.ml
index 7098a29..203e324 100644
--- a/lib/matchers/alpha/matcher.ml
+++ b/lib/matchers/alpha/matcher.ml
@@ -6,6 +6,7 @@ open Match
 open Range
 open Location
 open Types
+open Poly
 
 let configuration_ref = ref (Configuration.create ())
 let weaken_delimiter_hole_matching = false
diff --git a/lib/matchers/languages.ml b/lib/matchers/languages.ml
index 4101d11..a646154 100644
--- a/lib/matchers/languages.ml
+++ b/lib/matchers/languages.ml
@@ -1,6 +1,7 @@
 open Core
 
 open Types.Syntax
+open Poly
 
 let ordinary_string = Some { delimiters = [{|"|}]; escape_character = '\\' }
 
diff --git a/lib/matchers/types.ml b/lib/matchers/types.ml
index 1c260df..e0ba39c 100644
--- a/lib/matchers/types.ml
+++ b/lib/matchers/types.ml
@@ -1,4 +1,5 @@
 open Core
+open Poly
 
 module Syntax = struct
 
diff --git a/lib/rewriter/rewrite.ml b/lib/rewriter/rewrite.ml
index 629ce8a..5e3cbe2 100644
--- a/lib/rewriter/rewrite.ml
+++ b/lib/rewriter/rewrite.ml
@@ -2,6 +2,7 @@ open Core
 
 open Match
 open Replacement
+open Poly
 
 let debug =
   Sys.getenv "DEBUG_COMBY"
diff --git a/lib/server/server_types.ml b/lib/server/server_types.ml
index 6070d31..06ab55f 100644
--- a/lib/server/server_types.ml
+++ b/lib/server/server_types.ml
@@ -1,6 +1,7 @@
 open Core
 
 open Match
+open Poly
 
 module In = struct
 
diff --git a/src/benchmark.ml b/src/benchmark.ml
index 82a982f..8371545 100644
--- a/src/benchmark.ml
+++ b/src/benchmark.ml
@@ -1,4 +1,5 @@
 open Core
+open Poly
 
 module Time = Core_kernel.Time_ns.Span
 
diff --git a/src/main.ml b/src/main.ml
index 6b17a64..23f44d9 100644
--- a/src/main.ml
+++ b/src/main.ml
@@ -3,6 +3,7 @@ open Command.Let_syntax
 open Hack_parallel
 
 open Pipeline.Command_configuration
+open Poly
 
 let verbose_out_file = "/tmp/comby.out"
 
