commit e79e945bf2ae8daeb98a045c6d3b3e0b4ea4df11
Author: Matthew Glazar <strager.nds@gmail.com>
Date:   Tue Nov 19 22:06:02 2019 -0800

    Fix use of Lwt_unix

diff --git a/lib/pipeline/pipeline.ml b/lib/pipeline/pipeline.ml
index 00934e9..71e864a 100644
--- a/lib/pipeline/pipeline.ml
+++ b/lib/pipeline/pipeline.ml
@@ -381,7 +381,7 @@ let run
          starting up at module load. Lwt interferes with hack_parallel child
          workers otherwise. This command runs the setup function, only after we are
          finished with parallel work. *)
-      Lwt_unix.i_am_the_captain_now ();
+      (*Lwt_unix.i_am_the_captain_now ();*)
       let { editor; default_is_accept } = Option.value_exn interactive_review in
       Interactive.run editor default_is_accept count rewrites;
       count
